### ISSUE TITLE
ResourceLoader.has_cached() giving unexpected output when ResourceLoader.load() called with no_cache=true

### ISSUE THREAD

**Issue by AdwaithHSivam** [godotengine/godot#59752]
### Godot version

3.3.stable, 3.4.4.stable

### System information

Linux Mint 20.3

### Issue description

```gdscript
func _ready():
	var res = Resource.new()
	var path = "res://test.tres"
	ResourceSaver.save(path, res)
	for i in range(10):
		res = ResourceLoader.load(path, "", true) # no_cache=true
		print(ResourceLoader.has_cached(path))
```
This code gives the following output
```
True
False
True
False
True
False
True
False
True
False
```
It should be all False.

### Steps to reproduce

- Save a resource using `ResourceSaver.save()` at a path.
- Load it using `ResourceLoader.load()` with `no_cache = true`.
- Use `ResourceLoader.has_cached()` on the path.
- Repeat step 2 & 3 few times.

### Minimal reproduction project

[Minimal.zip](https://github.com/godotengine/godot/files/8391373/Minimal.zip)

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/59686.

**Comment by AdwaithHSivam** [comment]
Temporary workaround for anyone having the same problem:
use `res.resource_path = ""` or `res.take_over_path("")` after the load().
In case `res` has subresources do this for them too.

**Comment by Flavelius** [comment]
And another issue that is for a different method, but likely related as it is also based on cache (rebuild/refresh) misbehaviour: https://github.com/godotengine/godot/issues/43990

**Comment by akien-mga** [comment]
Fixed by #62408.
